crates/matrix-sdk-crypto/src/session_manager/sessions.rs

```diff
@@ -414,18 +414,23 @@ impl SessionManager {
                 .collect();
 
             if !missing_devices.is_empty() {
+                let mut missing_devices_by_user: BTreeMap<_, BTreeSet<_>> = BTreeMap::new();
+                for (user_id, device_id) in missing_devices.into_iter() {
+                    missing_devices_by_user
+                        .entry(user_id)
+                        .or_default()
+                        .insert((*device_id).clone());
```

Comment:
What's the type of `device_id` here that makes an explicit dereference necessary?

Reply:
it is `&&OwnedDeviceId`. 

Note that this explicit deref is not new; it's just moved. It was suggested by poljar in [this thread](https://github.com/matrix-org/matrix-rust-sdk/pull/2805#discussion_r1387901456).